src/utils/utils_cache_line.c

```diff
@@ -44,6 +45,7 @@ static void __set_cache_line_invalid(struct ocf_cache *cache, uint8_t start_bit,
 	 */
 	if (!is_valid && !ocf_cache_line_are_waiters(cache, line)) {
 		ocf_purge_eviction_policy(cache, line);
+		ocf_promotion_cache_line_purge(cache->promotion_policy, line);
```

Comment:
We should remove line from pp when it's being promoted, not evicted.

Reply:
Changed the locations for purge - it needed also to be in core line discard path